src/Databases/DatabaseFactory.cpp

```diff
@@ -432,6 +437,45 @@ DatabasePtr DatabaseFactory::getImpl(const ASTCreateQuery & create, const String
         return std::make_shared<DatabaseSQLite>(context, engine_define, create.attach, database_path);
     }
 #endif
+    else if (engine_name == "Filesystem")
+    {
+        const ASTFunction * engine = engine_define->engine;
+
+        /// If init_path is empty, then the current path will be used
+        std::string init_path;
+
+        if (engine->arguments && !engine->arguments->children.empty())
+        {
+            if (engine->arguments->children.size() != 1)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "Filesystem database requires at most 1 argument: filesystem_path");
+
+            const auto & arguments = engine->arguments->children;
+            init_path = safeGetLiteralValue<String>(arguments[0], engine_name);
+        }
+
+        return std::make_shared<DatabaseFilesystem>(database_name, init_path, context);
+    }
+#if USE_AWS_S3
+    else if (engine_name == "S3")
+    {
+        const ASTFunction * engine = engine_define->engine;
+
+        std::string key_id;
+        std::string secret_key;
+
+        if (engine->arguments && !engine->arguments->children.empty())
+        {
+            if (engine->arguments->children.size() != 2)
+                throw Exception(ErrorCodes::BAD_ARGUMENTS, "S3 database requires 0 or 2 argument: [access_key_id, secret_access_key]");
+
+            const auto & arguments = engine->arguments->children;
+            key_id = safeGetLiteralValue<String>(arguments[0], engine_name);
```

Comment:
please check https://github.com/ClickHouse/ClickHouse/blob/a8e63abbb48abd70026d71e5e0189f9efcbfeb28/src/Storages/StorageS3.cpp#L1288
it will be good to support features which are supported for ordinary S3 table engine, e.g. named collections and NOSIGN keyword,

Reply:
fixed

Added the following specifications for DatabaseS3 and support named collections
```
 - S3()
 - S3('url')
 - S3('url', 'NOSIGN')
 - S3('url', 'access_key_id', 'secret_access_key')
 ```
 The other options look unnecessary for database (`format`, for example)